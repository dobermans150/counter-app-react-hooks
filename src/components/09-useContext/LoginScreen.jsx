import React, { useContext } from 'react'
import { UserContext } from './userContext';

export const LoginScreen = () => {

    const { setUser } = useContext( UserContext );

    const userData = {
        id: 'use2134',
        name: 'Christian',
        email: 'christianjguevara@mail.com'
    }

    return (
        <div>
            <h1>Login Screen</h1>
            <hr />

            <button
                className="btn btn-primary"
                onClick={ () => setUser( userData ) }
            >
                Login
            </button>
        </div>
    )
}
