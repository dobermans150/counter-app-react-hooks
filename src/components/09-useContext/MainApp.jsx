import React, { useState } from 'react'
import { AppRouter } from './AppRouter'
import { UserContext } from './userContext'

export const MainApp = () => {
    /*    const user = {
           id: 'use2134',
           name: 'Christian',
           email: 'christianjguevara@mail.com'
       } */

    const [ user, setUser ] = useState( {} )

    return (

        <UserContext.Provider value={ { user, setUser } } >
            <AppRouter />
        </UserContext.Provider>

    )
}
