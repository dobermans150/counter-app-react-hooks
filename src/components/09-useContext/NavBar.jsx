import React from 'react'
import { Link, NavLink } from "react-router-dom";

export const NavBar = () => {
    return (
        <nav className="navbar navbar-expand-lg navbar-light bg-light">
            <div className="container-fluid">
                <Link className="navbar-brand" to="/">UseContext</Link>

                <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                    <div className="navbar-nav">
                        <NavLink activeClassName="active" className="nav-link " aria-current="page" to="/home">Home</NavLink>
                        <NavLink activeClassName="active" className="nav-link" to="/login">Login</NavLink>
                        <NavLink activeClassName="active" className="nav-link " to="/about">About</NavLink>
                    </div>
                </div>
            </div>
        </nav>
    )
}
