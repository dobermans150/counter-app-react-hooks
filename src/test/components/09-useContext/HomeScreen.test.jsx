import React from 'react';
import { mount } from 'enzyme'
import { HomeScreen } from '../../../components/09-useContext/HomeScreen';
import { UserContext } from '../../../components/09-useContext/userContext';

describe( 'pruebas en HomeScreen', () => {

    const user = {
        name: 'Fernando',
        email: 'correo@corre.com'
    }

    const wrapper = mount(
        <UserContext.Provider
            value={ user }
        >
            <HomeScreen />
        </UserContext.Provider>
    )

    test( 'Debe de mostrarse correctamente', () => {
        expect( wrapper ).toMatchSnapshot()
    } )


} )
