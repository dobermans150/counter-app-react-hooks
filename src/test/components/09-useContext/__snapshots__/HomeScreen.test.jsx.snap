// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en HomeScreen Debe de mostrarse correctamente 1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <pre
    className="container"
  />
</div>
`;
